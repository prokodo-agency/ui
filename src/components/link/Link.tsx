import { createIsland } from "@/helpers/createIsland"

import LinkServer from "./Link.server"

import type { LinkProps } from "./Link.model"

export const Link = createIsland<LinkProps>({
  name: "Link",
  Server: LinkServer,
  loadLazy: () => import("./Link.lazy"),
  // optional: custom predicate (e.g. always interactive if target="_blank")
  isInteractive: p => typeof p.linkComponent === "function",
})
