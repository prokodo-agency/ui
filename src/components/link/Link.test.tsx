import { render, screen } from "@/tests"

import { Link } from "./Link"

describe("The common link component", () => {
  it("should render primary", () => {
    render(
      <Link href="/" variant="primary">
        Primary link
      </Link>,
    )
    expect(screen.getByText("Primary link")).toMatchSnapshot()
  })

  it("should render secondary", () => {
    render(
      <Link href="/" variant="secondary">
        Secondary link
      </Link>,
    )
    expect(screen.getByText("Secondary link")).toMatchSnapshot()
  })

  it("should not open inbound links in a new tab", () => {
    render(
      <Link href="/" variant="primary">
        Inbound link
      </Link>,
    )
    const inbound = screen.getByText("Inbound link")
    expect(inbound).not.toHaveAttribute("rel")
    expect(inbound).not.toHaveAttribute("target")
  })
})
