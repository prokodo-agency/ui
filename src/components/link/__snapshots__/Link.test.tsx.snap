// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The common link component should render primary 1`] = `
<a
  class="Link"
  href="/"
>
  Primary link
</a>
`;

exports[`The common link component should render secondary 1`] = `
<a
  class="Link"
  href="/"
>
  Secondary link
</a>
`;
